src/azure-cli/azure/cli/command_modules/appconfig/_kv_import_helpers.py

```diff
@@ -0,0 +1,958 @@
+
+            elif format_ == "yaml":
+                for yaml_data in list(yaml.safe_load_all(config_file)):
+                    config_data.update(yaml_data)
+
+            elif format_ == "properties":
+                config_data = javaproperties.load(config_file)
+                logger.debug(
+                    "Importing feature flags from a properties file is not supported. If properties file contains feature flags, they will be imported as regular key-values."
+                )
+
+    return config_data
+
+
+def __read_features_from_file(file_path, format_):
+    config_data = {}
+    features_dict = {}
+
+    if format_ == "properties":
+        logger.warning(
+            "Importing feature flags from a properties file is not supported. If properties file contains feature flags, they will be imported as regular key-values."
+        )
+        return features_dict
+
+    try:
+        config_data = __read_with_appropriate_encoding(file_path, format_)
+        found_dotnet_schema = False
+        found_ms_fm_schema = False
+        dotnet_schema_keyword = None
+        for keywordset in (
+            FeatureManagementReservedKeywords.PASCAL,
+            FeatureManagementReservedKeywords.CAMEL,
+            FeatureManagementReservedKeywords.HYPHEN,
+        ):
+            # find the occurrences of feature management section in file.
+            if keywordset.feature_management in config_data:
+                if found_dotnet_schema:
+                    raise FileOperationError(
+                        'Unable to proceed because file contains multiple sections corresponding to "Feature Management".'
+                    )
+                features_dict[keywordset.feature_management] = config_data[
+                    keywordset.feature_management
+                ]
+                dotnet_schema_keyword = keywordset.feature_management
+                del config_data[keywordset.feature_management]
+                found_dotnet_schema = True
+
+        ms_feature_management_keyword = (
+            FeatureManagementReservedKeywords.UNDERSCORE.feature_management
+        )
+        if ms_feature_management_keyword in config_data:
+            if found_dotnet_schema and any(
+                key != FeatureFlagConstants.FEATURE_FLAGS_KEY
+                for key in config_data[ms_feature_management_keyword].keys()
+            ):
+                raise FileOperationError(
+                    "Data contains an already defined section with the key %s."
+                    % (dotnet_schema_keyword)
+                )
+
+            if (
+                FeatureFlagConstants.FEATURE_FLAGS_KEY
+                in config_data[ms_feature_management_keyword]
+            ):
+                found_ms_fm_schema = True
+
+            if any(
+                key != FeatureFlagConstants.FEATURE_FLAGS_KEY
+                for key in config_data[ms_feature_management_keyword].keys()
+            ):
```

Comment:
The same check is repeated here. Can you move it to a common local variable?

Reply:
Updated